Protocol.md

```diff
@@ -23,188 +44,182 @@ The following figure gives a brief overview of the protocol:
+across many distinct blockchains as well as have contracts within multiple
+blockchains opened to the same peer (if it supports the target chains).
 
-1. Funding Public Key
-   - A's public key used in the multi-signature [funding output](Transactions.md#FundingOutputs)
-2. Sweep Public Key
-   - A's public key used in the creation of the CETs [output script](Transactions.md#CETOutputs).
+`contract_info` specifies the sender's payouts for all events. `oracle_info`
+specifies the oracle(s) to be used as well as their commitments to events.
 
-#### A's addresses
+`funding_pubkey` is the public key in the 2-of-2 multisig script of
+the funding transaction output. `payout_spk` specifies the script
+pubkey that CETs and the refund transaction should use in the sender's output.
 
-The following addresses:
+`total_collateral_satoshis` is the amount the sender is putting into the
+contract. `num_funding_inputs` is the number of funding inputs contributed by
+the sender and `funding_inputs` contains outputs, outpoints, and expected weights
+of the sender's funding inputs. `change_spk` specifies the script pubkey that funding
+change should be sent to.
 
-1. Funding Change Address
-   - The address to use to send the change if the sum of A's inputs is greater than the fund amount plus the fees.
-2. Final Address
-   - The address to which funds will be sent in [unilateral contract execution](Transactions.md#CETOutputs), [refund](Transactions.md#refund-transaction) and [mutual closing](Transactions.md#mutual-closing-transaction) transactions.
+`feerate_per_vb` indicates the fee rate in satoshi per virtual byte that both
+sides will use to compute fees in the funding transaction, as described in the
+[transaction specification](Transactions.md).
 
-#### A's inputs
-The set of UTXOs to be used as input to the fund transaction.
+`contract_maturity_cltv` is the CLTV timelock to be put on CETs. `contract_timeout_cltv` is the CLTV timelock to be put on the refund transaction.
 
-#### Estimatesmartfee (satoshi/vbyte)
-The fee rate to be used when computing fees for the different transactions.
+#### Requirements
 
-#### CET CSV Delay
-The number to use as input to the CSV in the [CETs](Transactions.md#outputs-1).
+The sending node MUST:
 
-#### CET lock time
-The number to use as `nLockTime` for the CET transactions.
-This should correspond to the maturity date of the contracts, invalidating early CET broadcasting.
+  - set undefined bits in `contract_flags` to 0.
+  - ensure the `chain_hash` value identifies the chain it wishes to open the contract within.
+  - set `funding_pubkey` to a valid secp256k1 pubkey in compressed format.
+  - set `total_collateral_satoshis` greater than or equal to 1000.
+  - set `contract_maturity_cltv` and `contract_timeout_cltv` to either both be UNIX timestamps, or both be block heights as distinguished in [BIP 65](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki).
```

Comment:
Firstly, I'll note that this has come up in multiple spec meetings and we have always ended up agreeing that wall clock time is the only way to go for practical applications. (e.g. reasoning about the price of bitcoin in USD at a given block height is pretty silly)

Regardless this is not an actual security concern since this timelock is just to make UX and requirements better but in no exact way (e.g. they don't have to be online and also have time to update terms if an oracle loses its signing keys).